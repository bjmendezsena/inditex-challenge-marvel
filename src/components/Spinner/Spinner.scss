@import "@/styles/variables.scss";

.load {
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  color: $primary-color;
  &__icon {
    animation: linear load 2s infinite;
    width: 32px;
    height: 32px;
    fill: currentColor;

    &-wrap {
      margin: auto;
    }

    @keyframes load {
      from {
        transform: rotate(0deg) scale(2);
      }
      to {
        transform: rotate(360deg) scale(2);
      }
    }
  }
}
