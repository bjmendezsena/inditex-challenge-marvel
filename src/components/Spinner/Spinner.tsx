import { ComponentProps } from 'react';
import './Spinner.scss';

export type SpinnerProps = ComponentProps<'div'>;

export const Spinner = (props: SpinnerProps) => {
  return (
    <div className="load" {...props}>
      <div className="load__icon-wrap">
        <svg className="load__icon">
          <path
            fill="currentColor"
            d="M12,4V2A10,10 0 0,0 2,12H4A8,8 0 0,1 12,4Z"
          />
        </svg>
      </div>
    </div>
  );
};
