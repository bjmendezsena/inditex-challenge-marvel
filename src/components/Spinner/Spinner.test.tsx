import { render } from '@testing-library/react';
import { Spinner } from './Spinner';

describe(`<${Spinner.name} />`, () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<Spinner />);

    expect(asFragment()).toMatchSnapshot();
  });
});
