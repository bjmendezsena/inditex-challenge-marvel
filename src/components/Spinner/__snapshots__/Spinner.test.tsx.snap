// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="load"
  >
    <div
      class="load__icon-wrap"
    >
      <svg
        class="load__icon"
      >
        <path
          d="M12,4V2A10,10 0 0,0 2,12H4A8,8 0 0,1 12,4Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
